ipam/status.go

```diff
@@ -0,0 +1,54 @@
+package ipam
+
+import (
+	"github.com/weaveworks/weave/ipam/paxos"
+	"github.com/weaveworks/weave/net/address"
+)
+
+type Status struct {
+	Paxos         *paxos.Status
+	Range         string
+	DefaultSubnet string
+	Entries       []EntryStatus
+}
+
+type EntryStatus struct {
+	Token   string
+	Peer    string
+	Version uint32
+}
+
+func NewStatus(allocator *Allocator, defaultSubnet address.CIDR) *Status {
+	if allocator == nil {
+		return nil
+	}
+
+	resultChan := make(chan *Status)
+	allocator.actionChan <- func() {
+		resultChan <- &Status{
+			paxos.NewStatus(allocator.paxos),
+			allocator.universe.String(),
+			defaultSubnet.String(),
+			newEntryStatusSlice(allocator)}
+	}
+
+	return <-resultChan
+}
+
+func newEntryStatusSlice(allocator *Allocator) []EntryStatus {
+	var slice []EntryStatus
+
+	if allocator.ring.Empty() {
+		return slice
+	}
+
+	for _, entry := range allocator.ring.Entries {
+		slice = append(slice,
+			EntryStatus{
+				entry.Token.String(),
+				entry.Peer.String(),
+				entry.Version})
+	}
+
+	return slice
+}
```

Comment:
I'd put this in allocator.go since a) it's not that long, and b) it is rather deeply tied to the allocator implementation, including taking locks etc.